testing/endpoint.go

```diff
@@ -608,6 +608,33 @@ func (endpoint *Endpoint) ChanUpgradeTry(timeout channeltypes.Timeout) error {
 	return endpoint.Chain.sendMsgs(msg)
 }
 
+// ChanUpgradeAck sends a MsgChannelUpgradeAck to the associated endpoint.
+func (endpoint *Endpoint) ChanUpgradeAck() error {
+	counterpartyChannelID := endpoint.Counterparty.ChannelID
+	counterpartyPortID := endpoint.Counterparty.ChannelConfig.PortID
+
+	channelKey := host.ChannelKey(counterpartyPortID, counterpartyChannelID)
+	proofChannel, height := endpoint.Counterparty.Chain.QueryProof(channelKey)
+	upgradeKey := host.ChannelUpgradeKey(counterpartyPortID, counterpartyChannelID)
+	proofUpgrade, _ := endpoint.Counterparty.Chain.QueryProof(upgradeKey)
+
```

Comment:
Opened #3722, I noticed in #3671 we will likely copy/paste this code, so it might be nicer to bundle into helper functions. What do you think?

Reply:
Yeah, thought about this too but didn't get around opening an issue. Idea sounds good and I think there's a couple other `Query` funcs already so precedence!

Reply:
this can be updated now